information-schema/information-schema-slow-query.md

```diff
@@ -262,3 +262,60 @@ Currently, because statistics of the system tables are not collected, sometimes
 ```sql
 SELECT /*+ AGG_TO_COP() */ COUNT(*) FROM CLUSTER_SLOW_QUERY GROUP BY user;
 ```
+
+# Statistics
+
+When running a [`EXPLAIN ANALYZE`](/sql-statements/sql-statement-explain-analyze.md) query against the `SLOW_QUERY` table it will contain information about how the database fetched the slow query information. This data is not available when running `EXPLAIN ANALYZE` against the `CLUSTER_SLOW_QUERY` table.
+
+Example:
+
+```sql
+EXPLAIN ANALYZE SELECT * FROM information_schema.slow_query LIMIT 1\G
+```
+
+```
+*************************** 1. row ***************************
+            id: Limit_7
+       estRows: 1.00
+       actRows: 1
+          task: root
+ access object: 
+execution info: time:3.46ms, loops:2, RU:0.000000
+ operator info: offset:0, count:1
+        memory: N/A
+          disk: N/A
+*************************** 2. row ***************************
+            id: └─MemTableScan_10
+       estRows: 10000.00
+       actRows: 64
+          task: root
+ access object: table:SLOW_QUERY
+execution info: time:3.45ms, loops:1, initialize: 55.5µs, read_file: 1.21ms, parse_log: {time:4.11ms, concurrency:15}, total_file: 1, read_file: 1, read_size: 4.06 MB
+ operator info: only search in the current 'tidb-slow.log' file
+        memory: 1.26 MB
+          disk: N/A
+2 rows in set (0.01 sec)
+```
+
+The extra information here is this (formatted for readability):
+```
+initialize: 55.5µs,
+read_file: 1.21ms,
+parse_log: {
+  time:4.11ms,
+  concurrency:15
+},
+total_file: 1,
+read_file: 1,
+read_size: 4.06 MB
+```
+
+| Field | Description |
+|---|---|
+| `initialize` | Time it took to initialize |
+| `read_file` | Time spend reading file(s) |
+| `parse_log.time` | Time spend parsing the log |
+| `parse_log.concurrency` | Concurrency for parsing the log (set by [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) ) |
+| `total_file` | Number of files read |
+| `read_file` | Number of the file that was read |
+| `read_size` | Bytes read from the log files |
```

Comment:
```suggestion
| `initialize` | Time spent initializing |
| `read_file` | Time spent reading file(s) |
| `parse_log.time` | Time spent parsing the slow log file |
| `parse_log.concurrency` | Concurrency for parsing the slow log file (set by [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) ) |
| `total_file` | Number of files to read |
| `read_file` | Number of files that are read  |
| `read_size` | Bytes read from the log file |
```

Reply:
The suggested "to read" isn't correct. This is the number of files that actually were read before, not something that would look to the future

Reply:
`read_file` seems also the number of files that are read. What's the difference between `total_file` and `read_file`?

Reply:
`read_file` is `readFileNum` and it is increased in `getNextFile()`

https://github.com/pingcap/tidb/blob/64c8d318551e6ff43d5c01e6e46f22574003bb66/pkg/executor/slow_query.go#L177

Note that `read_file` is twice in the output, the first time it is the `time.Duration` in `readFile`.

And `total_file` is `totalFileNum` and is increased in `getAllFiles()`

https://github.com/pingcap/tidb/blob/64c8d318551e6ff43d5c01e6e46f22574003bb66/pkg/executor/slow_query.go#L968

So to me it looks like:
- `total_file` is the total number of log files found
- `read_file` is the number of files that was actually read

I would assume that `read_file <= total_file`.

Reading of files might stop early based on the filtering conditions for the time range specified in the query.

The struct that is used for this (unfortunately without useful comments):
https://github.com/pingcap/tidb/blob/64c8d318551e6ff43d5c01e6e46f22574003bb66/pkg/executor/slow_query.go#L1103-L1111

The code for formatting the output:
https://github.com/pingcap/tidb/blob/64c8d318551e6ff43d5c01e6e46f22574003bb66/pkg/executor/slow_query.go#L1113-L1119

Reply:
Many thanks for the detailed explanation.  I've updated my comments accordingly. PTAL again.